woocommerce-gateway-paylike/includes/class-wc-gateway-paylike.php

```diff
@@ -0,0 +1,557 @@
+     *
+     * @return bool
+     */
+    public function process_refund( $order_id, $amount = null, $reason = '' ) {
+        $order          = wc_get_order( $order_id );
+        $transaction_id = get_post_meta( $order_id, '_paylike_transaction_id', true );
+        if ( ! $order || ! $transaction_id ) {
+            return false;
+        }
+        $data = array();
+        if ( ! is_null( $amount ) ) {
+            $data['amount'] = $this->get_paylike_amount( $amount );
+        }
+
+        WC_Paylike::log( "Info: Beginning refund for order $order_id for the amount of {$amount}" );
+        $captured = get_post_meta( $order_id, '_paylike_transaction_captured', true );
+        if ( 'yes' == $captured ) {
+            if ( $reason ) {
+                $data['descriptor'] = $reason;
+            }
+            $result = Paylike\Transaction::refund( $transaction_id, $data );
+        } else {
+            $result = Paylike\Transaction::void( $transaction_id, $data );
+        }
+
+        return $this->handle_refund_result( $order, $result, $captured );
+
+    }
+
+    /**
+     * Display the pay button on the receipt page.
+     *
+     * @param $order_id
+     */
+    public function receipt_page( $order_id ) {
+        $order  = wc_get_order( $order_id );
+        $amount = $this->get_paylike_amount( $order->order_total, $order->get_order_currency() );
+        echo '<p>' . __( 'Thank you for your order, please click below to pay and complete your order.', 'woocommerce-gateway-paylike' ) . '</p>';
+        ?>
+        <button onclick="pay();"><?php _e( 'Pay Now', 'woocommerce-gateway-paylike' ); ?></button>
+        <script src="https://sdk.paylike.io/3.js"></script>
+        <script>
+            var paylike = Paylike('<?php echo $this->public_key;?>');
+
+            function pay() {
+
+                paylike.popup({
+                    title: '<?php echo get_bloginfo( 'name' ); ?>',
+                    currency: '<?php echo get_woocommerce_currency() ?>',
+                    amount:  <?php echo $amount; ?>,
+                    locale: '<?php echo get_locale(); ?>',
+                    custom: {
+                        orderNo: '<?php echo $order->get_order_number() ?>',
+                        email: '<?php echo $order->billing_email ?>',
+                        name: '<?php echo $order->billing_first_name . ' ' . $order->billing_last_name ?>',
+                        totalTax: '<?php echo $order->get_total_tax()?>',
+                        totalShipping: '<?php echo $order->get_total_shipping()?>',
+                        customerIP: '<?php echo $this->get_client_ip() ?>',
+                    },
+                }, function (err, res) {
+                    if (err)
+                        return console.warn(err);
+
+                    var trxid = res.transaction.id;
+                    jQuery("#complete_order").append('<input type="hidden" name="transaction_id" value="' + trxid + '" /> ');
```

Comment:
This is done on the checkout page where jQuery is used by woocommerce so yes, its safe but we can include it just to make this look more robust.